### ISSUE TITLE
Import: URIs containing un-escaped \ behave differently on Windows, Unix

### ISSUE THREAD

**Issue by scurest** [KhronosGroup/glTF-Blender-IO#1833]
Some misbehaving glTF producers (eg. assimp) may produce invalid URIs containing `\`. When imported, backslashes pass through `urllib.parse.unquote` unchanged and get passed on through `open` to the OS. On Windows the `\` will work as a folder separator and the URI will silently work. On Unix it will be treated as a normal filename character and (probably) fail.

The behavior should be the same on both OSes.

Note that the glTF validator does not flag URIs with a backslash as invalid. It treats it as though it had been properly percent-encoded, `%5C`.

#### Solution

I think the URI should be repaired by replacing backslashes with forward slashes, ie `uri = uri.replace('\\', '/')` before unquoting. The reasons:

1. This is what browsers do when they find a backslash. It's what the WHATWG's URL spec says to do.
2. Backslashes are almost certain to be improperly encoded Windows folder separators (they are not common in Unix filenames).

The alternative is to raise an error and refuse to use the path.

Also the `_uri_to_path` function used for loading images should be moved out so it can be used for .bins as well. The change would go there.

**Comment by julienduroure** [comment]
Hello,
Do you have an example of glTF that have different behaviour on linux and Windows?
Using linux here:

I just tried to modify one with 1 `\`, but of course the json library complains that json is not valid (on linux again)

![image](https://user-images.githubusercontent.com/1696020/213799349-be835d72-cfa8-47ff-ae05-cf185ba4006f.png)

Testing with `\\`, and importer works without any error (on linux again)

![image](https://user-images.githubusercontent.com/1696020/213799727-13424620-d5d6-4df2-8996-164111386bff.png)

**Comment by scurest** [comment]
Try a .bin. Images go through Blender's own file loading code, which will treat `\` as `/`.

```
    "buffers" : [
        {
            "uri" : "data\\somefile.bin"
        }
    ]
```

**Comment by julienduroure** [comment]
Proposition of fix in #1834
